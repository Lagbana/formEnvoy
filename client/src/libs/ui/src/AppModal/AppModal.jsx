import React from 'react';
import { Platform } from 'react-native';
import { AppModal as AppModalNative } from './components/AppModal.native';
import { AppModal as AppModalWeb } from './components/AppModal.web';

export const AppModal: React.FC<any> = (props) => {
  switch (Platform.OS) {
    case 'ios':
    case 'android':
      return <AppModalNative {...props} />;
    default:
      return <AppModalWeb {...props} />;
  }
};
