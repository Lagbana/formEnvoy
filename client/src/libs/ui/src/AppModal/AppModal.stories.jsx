import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppModal } from './AppModal'

storiesOf('AppModal', module)
  .add('Visible: True', () => (
    <AppModal
			animationType="slide"
			visible={true}
			presentationStyle="pageSheet"
			onShow={() => null}
			onDismiss={() => null}
			onRequestClose={() => {}}
    />
  ))
  .add('Visible: False', () => (
    <AppModal
			animationType="slide"
			visible={false}
			presentationStyle="pageSheet"
			onShow={() => null}
			onDismiss={() => null}
			onRequestClose={() => { }}
    />
  ))
