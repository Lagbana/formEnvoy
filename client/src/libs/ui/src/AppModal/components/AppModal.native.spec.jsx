import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppModal } from './AppModal.native'

configure({ adapter: new Adapter() })

test('<AppModal /> renders correctly', () => {
  const component = shallow(<AppModal visible={true} />)
  component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})
