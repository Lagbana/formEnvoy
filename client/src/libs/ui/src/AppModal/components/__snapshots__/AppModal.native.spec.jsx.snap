// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppModal /> renders correctly 1`] = `""`;
